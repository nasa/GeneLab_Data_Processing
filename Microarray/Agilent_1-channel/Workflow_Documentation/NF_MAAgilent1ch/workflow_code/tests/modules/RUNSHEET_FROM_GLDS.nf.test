nextflow_process {

    name "Test Process RUNSHEET_FROM_GLDS"
    script "modules/RUNSHEET_FROM_GLDS.nf"
    process "RUNSHEET_FROM_GLDS"

    test("GLDS-367: Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                biomart_attribute = "agilent_wholegenome_4x44k_v1"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "OSD-367"
                input[1] = "GLDS-367"
                input[2] = file("${baseDir}/bin/dp_tools__agilent_1_channel")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
