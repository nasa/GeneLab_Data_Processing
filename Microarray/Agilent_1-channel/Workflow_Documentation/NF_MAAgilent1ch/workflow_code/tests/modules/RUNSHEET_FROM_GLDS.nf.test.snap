{
    "GLDS-367: Should run without failures": {
        "content": [
            {
                "runsheet": [
                    "OSD-367_microarray_v0_runsheet.csv:md5,6b066905180d502a70e4ad9fc094433d"
                ],
                "isaArchive": [
                    "GLDS-367_metadata_GLDS-367-ISA.zip:md5,130bb4f9fded2750631cfed98c173c06"
                ]
            }
        ],
        "timestamp": "2023-02-15T00:16:07+0000"
    }
}